Dockerfile

```diff
@@ -1,24 +1,46 @@
-FROM ruby:2.7.2
-RUN apt-get update -qq && apt-get upgrade -y
-RUN apt-get install -y build-essential nodejs && apt-get clean
-RUN gem install foreman
+ARG base_image=ruby:2.7.2
 
-ENV GOVUK_APP_NAME government-frontend
-ENV PORT 3090
-ENV RAILS_ENV development
+FROM $base_image AS builder
 
-ENV APP_HOME /app
-RUN mkdir $APP_HOME
+ENV RAILS_ENV=production
 
-WORKDIR $APP_HOME
-ADD Gemfile* $APP_HOME/
-ADD .ruby-version $APP_HOME/
-RUN bundle install
+RUN apt-get update -qy && \
+    apt-get upgrade -y && \
+    apt-get install -y build-essential nodejs && \
+    apt-get clean 
+    
+RUN bundle config set force_ruby_platform true
 
-ADD . $APP_HOME
+RUN mkdir /app
 
-RUN GOVUK_WEBSITE_ROOT=https://www.gov.uk GOVUK_APP_DOMAIN=www.gov.uk RAILS_ENV=production bundle exec rails assets:precompile
+WORKDIR /app
+
+COPY Gemfile Gemfile.lock .ruby-version /app/
+
+RUN bundle config set deployment 'true' && \
+    bundle config set without 'development test' && \
+    bundle install --jobs 4 --retry=2
+
+COPY . /app
+
+RUN GOVUK_APP_DOMAIN=www.gov.uk \
+    GOVUK_WEBSITE_ROOT=https://www.gov.uk \
+    bundle exec rails assets:precompile
```

Comment:
Precompiling assets into the image is another temporary workaround which we should signpost with a TODO (see https://github.com/alphagov/frontend/blob/main/Dockerfile). The reason it's important to comment those things is that a lot of the devs will be new to containers so it'll be helpful to distinguish for them between the things we're doing that are good practice and the things which are sketchy workarounds 😅